build.gradle

```diff
@@ -59,57 +60,81 @@ task configureGroovyAll {
-                        extensionModule.load(input)
-                    }
-                    extensionClasses << extensionModule.extensionClasses
-                    staticExtensionClasses << extensionModule.staticExtensionClasses
-                }
-
-                from(extractedFiles) {
-                    exclude { it.path == "META-INF/groovy/org.codehaus.groovy.runtime.ExtensionModule" }
+        def toMerge = [
+            'LICENSE'                       : 'META-INF/LICENSE',
+            'NOTICE'                        : 'META-INF/NOTICE',
+            'AST_TRANSFORMATION'            : 'META-INF/services/org.codehaus.groovy.transform.ASTTransformation',
+            'EXTENSION_MODULE'              : 'META-INF/groovy/org.codehaus.groovy.runtime.ExtensionModule',
+            'GROOVY_RELEASE_INFO_PROPERTIES': 'META-INF/groovy-release-info.properties',
+            'GROOVY_RUNNER'                 : 'META-INF/services/org.apache.groovy.plugin.GroovyRunner',
+            'INDEX_LIST'                    : 'META-INF/INDEX.LIST',
+            'MANIFEST_MF'                   : 'META-INF/MANIFEST.MF'
+        ]
+
+        def collectDuplicates = { File[] roots ->
+            Map<String, List<File>> duplicates = [:].withDefault { [] }
+            roots.each { root ->
+                root.eachFileRecurse(FileType.FILES) {
+                    def relative = root.toURI().relativize(it.toURI())
+                    duplicates[relative.toString()] += it
                 }
-                manifest.from extractedFiles.find { it.name == "MANIFEST.MF" }
             }
+            return duplicates.findAll { it.value.size() > 1 }
+        }
 
-            def extensionsModuleFile = file("${buildDir}/tmp/org.codehaus.groovy.runtime.ExtensionModule")
+        def merge = { String relative, List<File> files ->
+            switch (relative) {
+                case [toMerge.LICENSE, toMerge.NOTICE, toMerge.AST_TRANSFORMATION]:
+                    return files*.text.join("\n")
+                case [toMerge.GROOVY_RELEASE_INFO_PROPERTIES]:
+                    return files[0].text
+                case toMerge.EXTENSION_MODULE:
+                    return """
+                        moduleName=groovy-all
+                        moduleVersion=${(files[0].text =~ /moduleVersion=(.+)/)[0][1]}
+                        extensionClasses=${(files*.text.join('\n') =~ /extensionClasses=(.+)/)*.getAt(1).toSet().join(',')}
+                        staticExtensionClasses=${(files*.text.join('\n') =~ /staticExtensionClasses=(.+)/)*.getAt(1).toSet().join(',')}
+                    """.stripIndent().trim()
+                case [toMerge.MANIFEST_MF, toMerge.INDEX_LIST, toMerge.GROOVY_RUNNER]:
+                    return null
+                default:
+                    throw new IllegalThreadStateException(files.toString())
+            }
+        }
 
-            def extensnionModuleProperties = new Properties()
```

Comment:
previously the `ExtensionModule` was read, collected, new values created and new properties file created.
This is done inline in https://github.com/gradle/gradle-groovy-all/pull/5/files#diff-c197962302397baf3a4cc36463dce5eaR92 now.